src/Abstractions/NexusMods.Abstractions.Loadouts/Loadout.cs

```diff
@@ -1,150 +1,167 @@
     /// </summary>
-    public LoadoutKind LoadoutKind { get; init; }
-
-    /// <inheritdoc />
-    public override EntityCategory Category => EntityCategory.Loadouts;
-
+    public static readonly ULongAttribute Revision = new(Namespace, nameof(Revision));
+    
     /// <summary>
-    /// Summarises the changes made in this version of the loadout.
-    /// Think of this like a git commit message.
+    /// Retrieves all loadouts from the database.
     /// </summary>
-    public required string ChangeMessage { get; init; } = "";
-
-    /// <inheritdoc />
-    public static Loadout Empty(IDataStore store) => new()
+    public static IEnumerable<Model> Loadouts(this IDb db)
     {
-        LoadoutId = LoadoutId.Create(),
-        Installation = GameInstallation.Empty,
-        Name = "",
-        Mods = EntityDictionary<ModId, Mod>.Empty(store),
-        LastModified = DateTime.UtcNow,
-        PreviousVersion = EntityLink<Loadout>.Empty(store),
-        ChangeMessage = ""
-    };
+        return db.Find(Name)
+            .Select(db.Get<Model>);
+    }
 
     /// <summary>
-    /// Makes a change to the collection of mods stored.
+    /// Finds one or more loadouts by name.
     /// </summary>
-    /// <param name="modId">Unique identifier for this mod.</param>
-    /// <param name="func">Function used to change the details of this mod.</param>
-    /// <returns>A new loadout with the details of a single mod changed.</returns>
-    public Loadout Alter(ModId modId, Func<Mod, Mod?> func)
+    public static IEnumerable<Model> ByName(IDb db, string name)
     {
-        return this with
-        {
-            Mods = Mods.Keep(modId, func)
-        };
+        return db.FindIndexed(name, Name)
+            .Select(db.Get<Model>);
     }
 
     /// <summary>
-    /// Adds an individual mod to this loadout, returning a new loadout.
+    /// Gets all the revisions of a loadout over time
     /// </summary>
-    /// <param name="mod">An individual modification to add to this loadout.</param>
-    /// <returns>The loadout with this modification added.</returns>
-    public Loadout Add(Mod mod)
+    public static IObservable<Model> Revisions(this IConnection conn, LoadoutId id)
     {
-        return this with
-        {
-            Mods = Mods.With(mod.Id, mod)
-        };
+        // All db revisions that contain the loadout id, select the loadout
+        return conn.Revisions
+            .Where(db => db.Datoms(db.BasisTxId).Any(datom => datom.E == id.Value))
+            .StartWith(conn.Db)
+            .Select(db => db.Get<Model>(id.Value));
     }
 
     /// <summary>
-    /// Remove an individual mod from this loadout, returning a new loadout.
+    /// Returns all revisions of a loadouts over time.
     /// </summary>
-    /// <param name="mod"></param>
-    /// <returns></returns>
-    public Loadout Remove(Mod mod)
+    public static IObservable<Model> LoadoutRevisions(this IConnection conn)
```

Comment:
This returns an observable stream of all revisions of all loadouts, right? If that's the case, the `summary` should be updated, and the method should be named something like `AllLoadoutRevisions` or similar.

To prevent confusion between `LoadoutRevisions` and `Revisions` methods, we can also consider using names like `OneLoadoutAllRevisions` and `AllLoadoutsAllRevision` to make it clear that we're getting multiple revisions of one or all loadout(s).

Reply:
This was unused in the code, as I swapped over to IRepository<> later on in the PR. Removed.